app/core/api_client.py

```diff
@@ -17,15 +17,23 @@ class APIClient:
     def __init__(self, base_url: str, timeout: int = API_REQUEST_TIMEOUT):
         self.base_url = base_url
         self.timeout = timeout
-        self.client = httpx.AsyncClient(base_url=self.base_url, timeout=self.timeout,
-                                        verify=False)  # todo: check this verify
+        self.client = httpx.AsyncClient(
+            base_url=self.base_url, timeout=self.timeout, verify=False
+        )  # todo: check this verify
 
     async def _make_request(
-            self, method: str, endpoint: str, headers: dict[str, Any] | None = None,
-            params: dict[str, Any] | None = None,
-            data: dict[str, Any] | None = None
-    ) -> Response | dict[str, None | str | int] | dict[str, str | int | Any] | dict[str, None |
-                                                                                         str | int]:
+        self,
+        method: str,
+        endpoint: str,
+        headers: dict[str, Any] | None = None,
+        params: dict[str, Any] | None = None,
+        data: dict[str, Any] | None = None,
+    ) -> (
+        Response
+        | dict[str, None | str | int]
+        | dict[str, str | int | Any]
+        | dict[str, None | str | int]
+    ):
```

Comment:
You are more than welcome :)
The returned object is used to keep track of the error status and delegate the logic to the caller. I've found it very useful, to be honest, to have as many details of an error as I can. And yes, you can easily handle a 400 error differently from a 500 
It goes without saying that as a user of this class, you have to remember how to use it :) 
Why do you think it would be hard to customize the error handling by API Call? Even if the 3rd party APIs behaves differently, the `response` still has the `status_code` and it can be decoded as `json`